sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -146,6 +147,11 @@ object RapidsPluginUtils extends Logging {
     }
     // set driver timezone
     conf.set(RapidsConf.DRIVER_TIMEZONE.key, ZoneId.systemDefault().normalized().toString)
+
+    // We let spark.rapids.sql.multiThreadedRead.numThreads be same as spark.executor.cores.
+    // See: https://github.com/NVIDIA/spark-rapids/issues/5524
+    conf.set(RapidsConf.MULTITHREAD_READ_NUM_THREADS.key,
```

Comment:
I think we only set the default value as max(20, executor_cores) when the user doesn't set it explicitly. So, you have to check if it's set or not first.

Reply:
I think we want `RapidsConf.MULTITHREAD_READ_NUM_THREADS` to have a default of 0 (or auto). When it is set to 0, we try to derive it from the other configs.

That said, if the user sets it to something > 0, I'd expect us to honor that config, therefore we wouldn't overwrite it in this case.

Reply:
> Agree with @GaryShen2008. I think we want `RapidsConf.MULTITHREAD_READ_NUM_THREADS` to have a default of 0 (or auto). When it is set to 0, we try to derive it from the other configs.
> 
> That said, if the user sets it to something > 0, I'd expect us to honor that config, therefore we wouldn't overwrite it in this case.

That means: if we set default value of `RapidsConf.MULTITHREAD_READ_NUM_THREADS` as 0, then we must assign it a value from the user configs?

Reply:
Yes, you should get the MULTITHREAD_READ_NUM_THREADS first, if it's 0, which means a default value, then we'll use the max(20, executor-cores), otherwise, it means the user set it explicitly, then we don't need to change it.

Reply:
I think the default value of `MULTITHREAD_READ_NUM_THREADS ` should keep 20. 

If the user does not set both `spark.executor.cores` and `spark.rapids.sql.multiThreadedRead.numThreads`, then `MULTITHREAD_READ_NUM_THREADS` still has a valid value.

Reply:
The 0 helps you to understand if the user set the value or not.
If we keep 20, how do we know it's from user's setting or from the default value.

Reply:
> If we keep 20, how do we know it's from user's setting or from the default value.

We can judge this via `conf.contains("spark.rapids.sql.multiThreadedRead.numThreads")`. If the user set `spark.rapids.sql.multiThreadedRead.numThreads = 20` (same as the default value), `conf.contains("spark.rapids.sql.multiThreadedRead.numThreads")` will be true.